lib/improver/utilities/cube_manipulation.py

```diff
@@ -832,3 +833,36 @@ def enforce_coordinate_ordering(
     elif anchor == "end":
         cube.transpose(remaining_coords+coord_dims)
     return cube
+
+
+def clip_cube_data(cube, result):
+    """Apply np.clip to data in a resulting cube to ensure that the limits do
+    not go beyond the limits of the original input cube.
+
+    Args:
+        cube (iris.cube.Cube):
+            The cube from which data has been taken to be processed.
+        result (iris.cube.Cube):
+            The cube that has been processed and contains data that is to be
+            clipped.
+    Returns:
+        result_slices (iris.cube.Cube):
+            The processed cube with the data clipped to the limits of the
+            original preprocessed cube.
```

Comment:
and attributes and cell methods updated. Coordinate order check and reshuffled

Reply:
Yes, my mistake. It should be setting them back to the input "result" attributes and cell_methods, just in case they are lost in the slicing procedure. The reorder is likewise just to undo any changes caused by slicing.

Reply:
If we only need the min/max values from cube wouldn't it be better to supply these rather than a cube. I can see cases where we might want to clip the data (stop any probabilities <0 or > 1.0) outside of nbhooding where creating a cube just for this would be annoying.

Reply:
I agree. I have now changed this to simply take a minimum and maximum value. Tests updated.